server/test/services/CfJsonDocumentContextTest.java

```diff
@@ -428,40 +455,82 @@ public void readLong_returnsEmptyWhenTypeMismatch() {
   }
 
   @Test
-  public void readList_findsCorrectValue() {
+  public void readLongList_findsCorrectValue() {
     String testData = "{\"applicant\":{\"favorite_fruits\":[1, 2]}}";
     CfJsonDocumentContext data = new CfJsonDocumentContext(testData);
 
-    Optional<ImmutableList<Long>> found = data.readList(Path.create("applicant.favorite_fruits"));
+    Optional<ImmutableList<Long>> found =
+        data.readLongList(Path.create("applicant.favorite_fruits"));
 
     assertThat(found).hasValue(ImmutableList.of(1L, 2L));
   }
 
   @Test
-  public void readListWithOneValue_findsCorrectValue() {
+  public void readLongList_withOneValue_findsCorrectValue() {
     String testData = "{\"applicant\":{\"favorite_fruits\":[1]}}";
     CfJsonDocumentContext data = new CfJsonDocumentContext(testData);
 
-    Optional<ImmutableList<Long>> found = data.readList(Path.create("applicant.favorite_fruits"));
-
+    Optional<ImmutableList<Long>> found =
+        data.readLongList(Path.create("applicant.favorite_fruits"));
     assertThat(found).hasValue(ImmutableList.of(1L));
   }
 
   @Test
-  public void readList_pathNotPresent_returnsEmptyOptional() {
+  public void readLongList_pathNotPresent_returnsEmptyOptional() {
+    CfJsonDocumentContext data = new CfJsonDocumentContext();
+
+    Optional<ImmutableList<Long>> found = data.readLongList(Path.create("not.here"));
+
+    assertThat(found).isEmpty();
+  }
+
+  @Test
+  public void readLongList_withTypeMismatch_returnsEmptyOptional() {
+    String testData = "{ \"applicant\": { \"object\": { \"age\": 12 } } }";
+    CfJsonDocumentContext data = new CfJsonDocumentContext(testData);
+
+    Optional<ImmutableList<Long>> found = data.readLongList(Path.create("applicant.object.name"));
```

Comment:
Is this really a type mismatch or just an alternate path?

Reply:
Good catch thanks!